src/EditorFeatures/Core/Implementation/ExtractInterface/AbstractExtractInterfaceCommandHandler.cs

```diff
@@ -77,7 +77,7 @@ public bool ExecuteCommand(ExtractInterfaceCommandArgs args, CommandExecutionCon
                 }
 
                 var navigationService = workspace.Services.GetService<IDocumentNavigationService>();
-                navigationService.TryNavigateToPosition(workspace, result.NavigationDocumentId, 0);
+                navigationService.TryNavigateToPosition(workspace, result.NavigationDocumentId, 0, CancellationToken.None);
```

Comment:
Why not passing the cancellation token from the context? I can imagine you're wanting to force navigation once we already did the edit, but if so we probably should be also disabling cancellation before this line.

Reply:
this is a location where we do: `context.OperationContext.TakeOwnership();` so that we can show our own UI.  At that point, we're effectively not using a TWD anymore, right?

Reply:
Oh right. Thanks for adding a comment since that makes it actually clear now.